src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -97,6 +100,27 @@ public boolean doesModuleAlreadyExist(Module module, ObservableList<Module> modu
         return false;
     }
 
+    /**
+     * Checks if a Module is a valid preclusion.
+     * @param module Module to check.
+     * @param modules Module list to check against.
+     * @param preclusionMap Map that contains pairings of valid preclusions.
+     * @return True if module is a preclusion, false otherwise.
+     */
+    public boolean isModuleAPreclusion(Module module, ObservableList<Module> modules,
+                                       Map<String, String> preclusionMap) {
+        for (Module mod : modules) {
+            String currentModName = mod.getModuleCode().toString();
+            String modToCheckAgainst = module.getModuleCode().toString();
+            if (preclusionMap.containsKey(currentModName)) {
```

Comment:
We don't actually need to check `containsKey` HAHA if the map doesn't contain the key then the modules won't be equal (the next line)

Reply:
I left it out initially but it produced a nullptrexception,when mods that arent inside the map are checked against the map